src/Handlers/PaymentResponseHandler.php

```diff
@@ -251,31 +267,68 @@ public function handleAdyenApis(
             case self::REFUSED:
             case self::ERROR:
                 return [
-                        "isFinal" => true,
-                        "resultCode" => $this->paymentResponseHandlerResult->getResultCode(),
-                    ];
+                    "isFinal" => true,
+                    "resultCode" => $this->paymentResponseHandlerResult->getResultCode(),
+                ];
             case self::REDIRECT_SHOPPER:
             case self::IDENTIFY_SHOPPER:
             case self::CHALLENGE_SHOPPER:
             case self::PRESENT_TO_SHOPPER:
+            case self::PENDING:
                 return [
-                        "isFinal" => false,
-                        "resultCode" => $this->paymentResponseHandlerResult->getResultCode(),
-                        "action" => $this->paymentResponseHandlerResult->getAction()
-                    ];
+                    "isFinal" => false,
+                    "resultCode" => $this->paymentResponseHandlerResult->getResultCode(),
+                    "action" => $this->paymentResponseHandlerResult->getAction()
+                ];
                 break;
             case self::RECEIVED:
                 return [
-                        "isFinal" => true,
-                        "resultCode" => $this->paymentResponseHandlerResult->getResultCode(),
-                        "additionalData" => $this->paymentResponseHandlerResult->getAdditionalData()
-                    ];
+                    "isFinal" => true,
+                    "resultCode" => $this->paymentResponseHandlerResult->getResultCode(),
+                    "additionalData" => $this->paymentResponseHandlerResult->getAdditionalData()
+                ];
                 break;
             default:
                 return [
-                        "isFinal" => true,
-                        "resultCode" => self::ERROR,
-                    ];
+                    "isFinal" => true,
+                    "resultCode" => self::ERROR,
+                ];
         }
     }
+
+    /**
+     * Validates if the state is already changed where the resultCode would switch it
+     * Example: Authorised -> paid, Refused -> failed
+     *
+     * @param string $transactionStateTechnicalName
+     * @param string $resultCode
+     * @return bool
+     */
+    private function isTransactionHandled(
+        $transactionStateTechnicalName,
+        $resultCode
+    ) {
+        // TODO check all the states and adyen resultCodes not just the straightforward ones
```

Comment:
Let's create a follow up ticket for this

Reply:
PW-3121